### PR TITLE
Add POC for job submission via the OpenAPI Java SDK

### PR DESCRIPTION
**Pull request by abhi18av**
Hi team, 

This PR adds the first practical usage of the generated Java SDK to submit Jobs.

- The overall experience when failures happen is a bit raw i.e. minimal pre-flight request validation
- The optional parameters can be simply an empty string `""` but this would need to be addressed as we progress with the design

Also @matthdsm , since we want to use the OpenAPI SDK, I have removed most of the code which was copied from Github and added previously, which relied upon the deprecated/unmaintained Java SDK https://github.com/hashicorp/nomad-java-sdk. 

The screenshot below shows the successful submission from the SDK

image

### CODE REVIEW

File: plugins/nf-nomad/src/test/nextflow/nomad/executor/NomadClientTest.groovy [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def DEV_NOMAD_BASE_PATH = System.getenv("DEV_NOMAD_BASE_PATH")
+
+    def 'should create a client and submit a job'() {
+
+        given:
+        def defaultClient = Configuration
+                .getDefaultApiClient()
+                .setBasePath(DEV_NOMAD_BASE_PATH)
+
+
```
Comment (abhi18av) [inline plugins/nf-nomad/src/test/nextflow/nomad/executor/NomadClientTest.groovy:44]: The POC is done as a test based iteration.

File: settings.gradle [hunk 2]
```diff
@@ -17,3 +17,5 @@
 rootProject.name = 'nf-nomad'
 include 'plugins'
 include('plugins:nf-nomad')
+
+includeBuild('_resources/nextflow')
```
Comment (abhi18av) [inline settings.gradle:21]: This assumes that the main nextflow codebase is within the `/path/to/nf-nomad/_resources/nextflow` folder

File: plugins/nf-nomad/src/main/nextflow/nomad/config/NomadJobOpts.groovy [hunk 3]
```diff
@@ -23,4 +23,8 @@ package nextflow.nomad.config
  */
 
 class NomadJobOpts {
+
+    NomadJobOpts(Map config) {
+    }
+
```
Comment (abhi18av) [inline plugins/nf-nomad/src/main/nextflow/nomad/config/NomadJobOpts.groovy:29]: Initial attempt at modelling the config scope for nomad.

**Comment by matthdsm** [comment]
I'm not versed enough in anything Nomad to give a good review. Perhaps @tomiles or @jhaezebr will be able to give more info. 

Alternatively, if this is blocking your ability to continue hacking on this, feel free to hit the merge button.

**Review by matthdsm — APPROVED** [review]
See comment, feel free to merge if needed

**Comment by abhi18av** [comment]
Cool, for the time being moving ahead and when the team is back from holidays we can discuss and accommodate any design changes.

**Pull request merged**